### PR TITLE
Logging Infrastructure + Logging and SD Test Updates

### PR DESCRIPTION
**Pull request by yashika-batra**

### CODE REVIEW

File: state_machine/applications/system-check/tests/logging_infrastructure_test.py [hunk 1]
```diff
@@ -1,115 +1,126 @@
-from logging import clear_all_storage, get_buffer, log
+"""
```
Comment (jbwillis) [inline state_machine/applications/system-check/tests/logging_infrastructure_test.py:1]: delete header

File: state_machine/applications/system-check/main.py [hunk 2]
```diff
@@ -28,13 +29,15 @@
     - if it is to be run in default mode
 """
 all_tests = [
-    ("SD Test", "sd", tests.sd_test, True),
+    ("SD Test / Clear Logs", "sd", tests.sd_test, True),
```
Comment (jbwillis) [inline state_machine/applications/system-check/main.py:32]: Let's keep the SD test simple - no data deletion

File: state_machine/applications/system-check/tests/sd_test.py [hunk 3]
```diff
@@ -8,6 +8,7 @@ def sd_test():
     read from a file, delete file and test existence. Return result_dict
     values accordingly
     """
+    cubesat.clear_logs()
```
Comment (jbwillis) [inline state_machine/applications/system-check/tests/sd_test.py:11]: delete

File: state_machine/applications/system-check/tests/sd_test.py [hunk 4]
```diff
@@ -53,6 +54,8 @@ def sd_test():
     if filepath in os.listdir("/sd/"):
         return ("File deletion failed.", False)
 
+    cubesat.clear_logs()
```
Comment (jbwillis) [inline state_machine/applications/system-check/tests/sd_test.py:57]: delete

**Review by jbwillis — CHANGES_REQUESTED** [review]

**Review by thetazero — CHANGES_REQUESTED** [review]
There's a lot of driver code mixing with application level code.

Could you please try and do this with no modifications to the driver.

**Comment by jbwillis** [comment]
Made obsolete by #347

**Pull request closed without merging**